packages/nightingale-linegraph-track/src/nightingale-linegraph-track.js

```diff
@@ -0,0 +1,197 @@
+
+class NightingaleLinegraphTrack extends ProtvistaTrack {
+  connectedCallback() {
+    super.connectedCallback();
+
+    this._height = Number(this.getAttribute("height")) || 40;
+    this._yScale = scaleLinear();
+
+    if (this._data) this._createTrack();
+  }
+
+  set data(data) {
+    this._data = data;
+    this._createTrack();
+  }
+
+  _createTrack() {
+    select(this).selectAll("svg").remove();
+    this.svg = select(this)
+      .append("svg")
+      .attr("width", this.width)
+      .attr("height", this._height);
+    this.trackHighlighter.appendHighlightTo(this.svg);
+
+    const range = [];
+    const minimum = [];
+    const maximum = [];
+    this._data.forEach((d) => {
+      range.push(...d.range);
+      minimum.push(min(d.values, (v) => v.position));
+      maximum.push(max(d.values, (v) => v.position));
+    });
+    this.minRange = min(range);
+    this.maxRange = max(range);
+
+    this.beginning = min(minimum);
+    this.end = max(maximum);
+
+    // Create the visualisation here
+    this.chartGroup = this.svg.append("g").attr("class", "chart-group");
+    this._initYScale();
+
+    this.chart = this.chartGroup
+      .selectAll(".chart")
+      .data(this._data)
+      .enter()
+      .append("g")
+      .attr("class", "chart");
+
+    this.chart
+      .append("path")
+      .attr("class", "graph")
+      .attr("id", (d) => d.name)
+      .attr("d", (d) => {
+        d.colour = d.colour || interpolateRainbow(Math.random()); // eslint-disable-line no-param-reassign
+        return this.drawLine(d)(d.values);
+      })
+      .attr("fill", "none")
+      .attr("stroke", (d) => d.colour)
+      .attr("transform", "translate(0,0)");
+
+    const mouseG = this.chartGroup
+      .append("g")
+      .attr("class", "mouse-over-effects");
+
+    const lines = this.chartGroup.selectAll("path.graph");
+
+    const mousePerLine = mouseG
+      .selectAll(".mouse-per-line")
+      .data(this._data)
+      .enter()
+      .append("g")
+      .attr("class", "mouse-per-line");
+
+    mousePerLine
+      .append("circle")
+      .attr("r", 7)
+      .style("stroke", (d) => {
+        return d.colour;
+      })
+      .style("fill", "none")
+      .style("stroke-width", "1px")
+      .style("opacity", "0");
+
+    mousePerLine.append("text").attr("transform", "translate(10,3)");
+
+    const _this = this;
+    mouseG
+      .append("rect") // append a rect to catch mouse movements on canvas
+      .attr("width", this.width) // can't catch mouse events on a g element
+      .attr("height", this._height)
+      .attr("fill", "none")
+      .attr("pointer-events", "all")
+      .on("mouseout", () => {
+        // on mouse out hide circles and text
+        d3.selectAll(".mouse-per-line circle").style("opacity", "0");
+        d3.selectAll(".mouse-per-line text").style("opacity", "0");
+      })
+      // .on('mouseover', function() { // on mouse in show circles and text
```

Comment:
Did you want to leave this commented out on purpose for later, or can it be removed?

Reply:
It has to be removed. Mousemove does the job. It is redundant